src/zauction.ts

```diff
@@ -75,6 +126,15 @@ export function handleBuyNowPriceSet(event: BuyNowPriceSet): void {
   priceSet.save();
 }
 
+export function handleBuyNowPriceSetV2(event: BuyNowPriceSetV2): void {
```

Comment:
This is more of a retroactive comment, but you're capturing `BuyNowPriceSet`'s as events and not entities.

It would be more useful if instead of `BuyPriceSet` we have `BuyNowListing` where the id is the token id and you just create/update that entity whenever an event comes in.